app/src/main/java/com/futurice/freesound/feature/search/DefaultSearchDataModel.java

```diff
@@ -26,42 +26,81 @@
 
 import io.reactivex.Completable;
 import io.reactivex.Observable;
+import io.reactivex.functions.Action;
+import io.reactivex.functions.Consumer;
 import io.reactivex.subjects.BehaviorSubject;
 import polanski.option.Option;
 
 import static com.futurice.freesound.utils.Preconditions.get;
+import static timber.log.Timber.e;
 
 final class DefaultSearchDataModel implements SearchDataModel {
 
     @NonNull
     private final FreeSoundSearchService freeSoundSearchService;
 
     @NonNull
-    private final BehaviorSubject<Option<List<Sound>>> lastResultsStream = BehaviorSubject.create();
+    private final BehaviorSubject<Option<List<Sound>>> lastResultsOnceAndStream =
+            BehaviorSubject.createDefault(Option.none());
+
+    @NonNull
+    private final BehaviorSubject<Option<Throwable>> lastErrorOnceAndStream =
+            BehaviorSubject.createDefault(Option.none());
 
     DefaultSearchDataModel(@NonNull final FreeSoundSearchService freeSoundSearchService) {
         this.freeSoundSearchService = get(freeSoundSearchService);
     }
 
     @Override
-    @NonNull
     public Completable querySearch(@NonNull final String query) {
         return freeSoundSearchService.search(get(query))
                                      .map(SoundSearchResult::results)
                                      .map(Option::ofObj)
-                                     .doOnSuccess(lastResultsStream::onNext)
-                                     .toCompletable();
+                                     .doOnSuccess(storeValueAndClearError())
+                                     .doOnError(storeError(query))
+                                     .toCompletable()
+                                     .onErrorComplete();
+    }
+
+    @Override
+    @NonNull
+    public Observable<Option<List<Sound>>> getSearchResultsOnceAndStream() {
+        return lastResultsOnceAndStream.hide();
```

Comment:
I think I'd rather keep this Scheduler-neutral, if there's nothing downstream. The consumer of this Observable can then choose how they want to schedule it. If there was some downstream processing happening internally to the ViewModel, then I'd agree e.g. -
```
  return lastResultsOnceAndStream.observeOn(Schedulers.computation())
  .map(z -> z + 1) 
  .filter(y -> y > 1);
```